proxy/BUILD.bazel

```diff
@@ -44,3 +44,46 @@ go_library(
     importpath = "github.com/enfabrica/enkit/proxy/credentials",
     visibility = ["//visibility:public"],
 )
+
+load("@io_bazel_rules_docker//go:image.bzl", "go_image")
+load("@io_bazel_rules_docker//container:container.bzl", "container_push", "container_image")
+
+container_image(
+    name = "enproxy-base",
+    base = "@go_image_base//image",
+
+    # enproxy needs to cache certificates in a directory so they survive
+    # across restarts.
+    #
+    # This cache is extremely important: "let's encrypt" limits the number
+    # of certificates it issues for a domain in a given week. If we end
+    # up in a restart loop of the proxy without the cache, we'd quickly
+    # exhaust this limit, and won't be able to get new certificates until
+    # the rate limit grants more quota.
+    volumes = ["/cache", "/config"],
+    env = {
+      "HOME": "/cache",
+    },
+)
+
+go_image(
+    name = "enproxy-image",
+    base = ":enproxy-base",
+    binary = ":enproxy",
+    visibility = ["//visibility:public"],
```

Comment:
...is visibility:public everywhere a gazelle-ism? Can we turn that off? If not, why public?